File: _Src/Freem/Freem.Web.Api.Public.Contracts/Records/UpdateRecordRequest.cs
using Freem.Entities.Abstractions.Relations.Collection;
using Freem.Entities.Activities.Identifiers;
using Freem.Entities.Records.Models;
using Freem.Entities.Tags.Identifiers;

namespace Freem.Web.Api.Public.Contracts.Records;

public sealed class UpdateRecordRequest
{
    public UpdateField<RecordName?>? Name { get; init; }
    public UpdateField<RecordDescription?>? Description { get; init; }
    
    public UpdateField<IReadOnlyRelatedEntitiesIdentifiersCollection<ActivityIdentifier>>? Activities { get; init; }
    public UpdateField<IReadOnlyRelatedEntitiesIdentifiersCollection<TagIdentifier>>? Tags { get; init; }

    public bool HasChanges()
    {
        return
            Name is not null ||
            Description is not null ||
            Activities is not null ||
            Tags is not null;
    }
}

File: _Src/Freem/Freem.Web/src/clients/models/records/UpdateRecordRequest.ts
import {UpdateField} from "../UpdateField";

export class UpdateRecordRequest {
  public readonly period?: UpdateField<string>;
  public readonly name?: UpdateField<string>;
  public readonly description?: UpdateField<string>;
  public readonly activities?: UpdateField<string[]>;
  public readonly tags?: UpdateField<string[]>;

  constructor(
      period: UpdateField<string>,
      name?: UpdateField<string>,
      description?: UpdateField<string>,
      activities?: UpdateField<string[]>,
      tags?: UpdateField<string[]>
  ) {
    this.period = period;
    this.name = name;
    this.description = description;
    this.activities = activities;
    this.tags = tags;
  }
}